syntax = "proto3";

message createMessage{
    int32 error = 1;
}
message deleteMessage{
    int32 error = 1;
    int32 index = 2;
    
}
message listMessage{
    int32 error = 1;
    string result = 2;
}
message countMessage{
    int32 error = 1;
    string url = 2;
    int32 result = 3;
}
message enumerateMessage{
    int32 error = 1;
    string url = 2;
    string result = 3;
}

service workerCreate{
    rpc workerCreate(createMessage) returns (createMessage) {}}
service  workerDelete{
    rpc workerDelete(deleteMessage) returns (deleteMessage){}}
service listWorkers{
    rpc listWorkers(listMessage) returns (listMessage) {}}
service countWords{
    rpc countWords(countMessage) returns (countMessage) {}}
service enumerateWords{
    rpc enumerateWords(enumerateMessage) returns (enumerateMessage) {}}
